boards/DCM/Src/App/states/App_InitState.c

```diff
@@ -1,34 +1,106 @@
+    App_CanTx_SetPeriodicSignal_TORQUE_COMMAND_INVR(
+        can_tx_interface,
+        App_CanMsgs_dcm_invr_command_message_torque_command_invr_encode(0.0f));
 }
 
 static void InitStateRunOnTick1Hz(struct StateMachine *const state_machine)
 {
-    App_AllStatesRunOnTick1Hz(state_machine);
+    App_SharedStatesRunOnTick1Hz(state_machine);
 }
 
 static void InitStateRunOnTick100Hz(struct StateMachine *const state_machine)
 {
-    App_AllStatesRunOnTick100Hz(state_machine);
+    App_SharedStatesRunOnTick100Hz(state_machine);
 
     struct DcmWorld *world = App_SharedStateMachine_GetWorld(state_machine);
-    struct DcmCanTxInterface *can_tx_interface = App_DcmWorld_GetCanTx(world);
+    struct DcmCanRxInterface *can_rx_interface = App_DcmWorld_GetCanRx(world);
+    struct ErrorTable *       error_table = App_DcmWorld_GetErrorTable(world);
+    struct InverterSwitches * inverter_switches =
+        App_DcmWorld_GetInverterSwitches(world);
 
-    App_CanTx_SetPeriodicSignal_TORQUE_REQUEST(can_tx_interface, 0.0f);
-
-    // No need for any safety checks, just run! (this is a demo)
+#ifdef DEBUG // Enter drive state directly from init state
     App_SharedStateMachine_SetNextState(state_machine, App_GetDriveState());
+    return;
+#endif
+
+    // Provide LV to inverters when both AIRS are closed and DC bus voltage ~
+    // 400 V
+    if (App_SharedStates_AreBothAIRsClosed(can_rx_interface) &&
+        !App_InverterSwitches_IsRightOn(inverter_switches))
+    {
+        App_InverterSwitches_TurnOnRight(inverter_switches);
+    }
+    if (App_SharedStates_AreBothAIRsClosed(can_rx_interface) &&
+        !App_InverterSwitches_IsLeftOn(inverter_switches))
+    {
+        App_InverterSwitches_TurnOnLeft(inverter_switches);
+    }
+
+    // Holds previous start switch position (true = UP/ON, false = DOWN/OFF)
+    // Initialize to true to prevent a false start
+    static bool prev_start_switch_position = true;
+    const bool  current_start_switch_position =
+        App_SharedStates_IsStartSwitchOn(can_rx_interface);
+    const bool was_start_switch_pulled_up =
+        !prev_start_switch_position && current_start_switch_position;
+    const struct State *next_state = App_GetInitState();
+
+    if (App_SharedErrorTable_HasAnyCriticalErrorSet(error_table) ||
+        App_SharedStates_HasInverterFaulted(can_rx_interface))
+    {
+        // Transition to fault state if an inverter has faulted or
+        // a critical error has been set
+        next_state = App_GetFaultState();
+    }
+    else if (
+        App_SharedStates_AreBothAIRsClosed(can_rx_interface) &&
+        App_InitState_IsBrakeActuated(can_rx_interface) &&
+        was_start_switch_pulled_up)
+    {
+        // Transition to drive state when start-up conditions are passed (see
+        // EV.10.4.3):
+        next_state = App_GetDriveState();
+    }
+
+    prev_start_switch_position = current_start_switch_position;
```

Comment:
any way to make this name shorter for the start switch?

Reply:
prev_start_switch_pos ?

Reply:
made prev_start_switch_pos and curr_start_switch_pos